projects/packages/jetpack-mu-wpcom/src/features/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-launchpad-navigator.php

```diff
@@ -37,10 +37,68 @@ public function register_routes() {
 					'callback'            => array( $this, 'get_navigator_data' ),
 					'permission_callback' => array( $this, 'can_access' ),
 				),
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'update_navigator_options' ),
+					'permission_callback' => array( $this, 'can_access' ),
+					'args'                => array(
+						'active_checklist_slug' => array(
+							'description'       => 'The slug of the checklist to set as active.',
+							'type'              => 'string',
+							'validate_callback' => array( $this, 'validate_checklist_slug_param' ),
+						),
+					),
+				),
 			)
 		);
 	}
 
+	/**
+	 * Validates that the argument sent to the active_checklist_slug parameter is a valid checklist slug or empty.
+	 *
+	 * @param string $value The value of the active_checklist_slug parameter.
+	 * @return bool
+	 */
+	public function validate_checklist_slug_param( $value ) {
+		if ( $value === null ) {
+			return true;
+		}
+
+		return is_string( $value ) && in_array( $value, $this->get_checklist_slug_enums(), true );
+	}
+
+	/**
+	 * Returns all available checklist slugs.
+	 * TODO: This function is used by both endpoints, we should move it somewhere common.
+	 *
+	 * @return array Array of checklist slugs.
+	 */
+	public function get_checklist_slug_enums() {
+		$checklists = wpcom_launchpad_checklists()->get_all_task_lists();
+		return array_keys( $checklists );
+	}
+
+	/**
+	 * Updates Launchpad navigator-related options and returns the result
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 */
+	public function update_navigator_options( $request ) {
+		$updated = array();
+		$input   = $request->get_json_params();
+
+		foreach ( $input as $key => $value ) {
+			switch ( $key ) {
+				case 'active_checklist_slug':
+					$updated = wpcom_launchpad_set_current_active_checklist( $input['active_checklist_slug'] );
```

Comment:
Should we not put this into a key/subvalue of `$updated`? I ask because the current code does two things:
1. We change the type of `$update` to be a boolean instead of an array
2. We make the `foreach` loop's behaviour a bit more complicated in cases where we have multiple `$key` values, as we could end up overwriting the type and any existing values when we move through the loop.